inheritance.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+var Animal = function() {
+  this.stepsWalked = 0;
+  this.walk = function(steps) {
+    this.stepsWalked += steps;
+    console.log("i am walking ");
+  }
+  return this;
+};
+
+function Dog() {
+  Animal.call(this);
+}
+
+Dog.prototype = Object.create(Animal.prototype);
+Dog.prototype.constructor = Dog;
+
+Dog.prototype.wagTail = function() {
+  for (var i = 1; i <= 10; i++) {
+    setTimeout(function timer() {
+      console.log('waggle');
+    }, i * 3000);
+  }
+}
+
+function Cat() {
+  Animal.call(this);
+}
+
+Cat.prototype = Object.create(Animal.prototype);
+Cat.prototype.constructor = Cat;
+
+Cat.prototype.reactToPetting = function() {
+  var number = Math.floor((Math.random() * 10) + 1);
```

Comment:
`number` is a better name, but doesn't this variable represent the cat's mood? variable names like 'mood,' 'moodOfCat,' 'catMood' would describe *why* the variable exists.